### DISCUSSION TITLE
auto format PRs

### DISCUSSION THREAD

**Discussion by debkanchan** [trunk-io/docs#306] [Feature Requests]
I get a lot of dependabot/renovate PRs. 60% of the time trunk check fails and yells incorrect formatting. I have pre-commit hook to auto format but that only works locally. I need trunk to auto format PRs also.

**Comment by debkanchan** [comment]
The offending file is usually package.json

**Comment by gewenyu99** [comment]
I think the main challenge to this, as we discussed with the team, is that it becomes a hot mess and really difficult for your human-generated PRs. The formatter ends up being run on every push and that can be really wasteful.

Do you want this to be enabled for specific types of PRs? Filtered by contributor? Title? Do you want this to be run manually?

**Reply by debkanchan** [reply]
All my human prs are auto formatted by trunk pre commit. I just need it for dependabot/renovate prs
